File: Data Structures & Algorithms/first-missing-positive/submission-2.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            nums[i] = nums[i] if nums[i] > 0 else 0
        print (nums)

        for i in nums:
            if abs(i)-1 >= 0 and abs(i)-1 < len(nums):
                if nums[abs(i)-1] == 0:
                    nums[abs(i)-1] = 1
                nums[abs(i)-1] = -1*(abs(nums[abs(i)-1]))
        print (nums)
        
        for i in range(1,len(nums)+1):
            if nums[i-1] == abs(nums[i-1]):
                return i
        return len(nums)+1
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const obj = {};
        for (const num of nums) {
            obj[num] = 1;
        }

        const countSeq = (num) => {
            let max = 0;
            while (obj[num]) {
                num++;
                max++;
            }
            return max;
        }

        let ans = 0;

        for (const num of nums) {
            ans = Math.max(ans,countSeq(num));
        }
        
        return ans;
    }
}
